### DISCUSSION TITLE
[PSD12] Exercise 3.9

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#176] [Esercizi (A-L)]
**Exercise 3.9** The _toggle (T) flip-flop_ has one input, $CLK$, and one output, $Q$. On each rising edge of $CLK$, $Q$ toggles to the complement of its previous value. Draw a schematic for a T flip-flop using a D flip-flop and an inverter.

**Comment by Elia-Belli** [comment]
**Premessa**: nel testo penso ci sia un errore, in un T flip-flop $Q$ diventa $\bar Q$ ad ogni fronte positivo di $CLK$ **in cui T = 1**, e non solamente ad ogni fronte positivo. Se $Q$ dipendesse solo da $Q$ precedente, l'input $T$ non sarebbe di alcun utilizzo.

Detto questo realizzerò il circuito rispettando la definizione del testo.
![T](https://user-images.githubusercontent.com/107944476/213871642-a4f15cac-e785-4b3d-8fa5-7725e8bdff57.jpg)
Come richiesto il **toggle-flipflop** ha 1 solo input $CLK$, un output $Q$ ed è costruito a partire da un **D-fliflop** e $1\ NOT$

**Comment by fraxb04** [comment]
@Elia-Belli 
sinceramente non saprei fare l'esercizio, dato che non capisco come si possa costruire un T Flip-Flop, con un D Flip-Flop ed un inverter.
Allego una foto con la tabella della verità del D Flip-Flop, con le freccette accanto all'output che indicano quello che dovrebbe essere (l'output stesso), con il NOT.
![image](https://user-images.githubusercontent.com/122573871/213866470-033acf22-b4a6-4819-b072-7660d7f01467.png)

**Comment by Elia-Belli** [comment]
@fraxb04 quello che chiede l'esercizio è realizzare un flipflop che ad ogni colpo di clock campiona $D = \bar Q$, non si tratta del TFF che conosciamo, quindi basta inserire l'output negato come input. Nel tuo disegno la porta $NOT$ è al contrario.
La tabella sarebbe la seguente
| CLK  | Q | Q' |
| - | - | - |
| 0 | 0 | 0 |
| 0 | 1 | 1 |
| 1 | 0 | 1 |
| 1 | 1 | 0 |

**Comment by fraxb04** [comment]
@Elia-Belli 
Ok grazie mille, penso di aver capito.
Comunque per quanto riguarda il disegno, lo avevo copiato identico al tuo poiché non avevo capito come si facesse (proprio con la porta al contrario)
